src/test/v2.0/advanced/fdc3.raiseIntent-Result.ts

```diff
@@ -0,0 +1,121 @@
+import { wait } from "../../../utils";
+import { RaiseIntentControl2_0, IntentResultType, IntentApp } from "./intent-support-2.0";
+
+const control = new RaiseIntentControl2_0();
+
+export default () =>
+  describe("fdc3.raiseIntent (Result)", () => {
+    const RaiseIntentVoidResult0secs = "(2.0-RaiseIntentVoidResult0secs) App A receives a void IntentResult";
+    it(RaiseIntentVoidResult0secs, async () => {
+      await control.listenForError();
+      const intentResolution = await control.raiseIntent("aTestingIntent", "testContextX");
+      control.validateIntentResolution("IntentAppAId", intentResolution);
+      let intentResult = await control.getIntentResult(intentResolution);
+      control.validateIntentResult(intentResult, IntentResultType.Void);
+    });
+
+    // THIS TEST DOESN'T MAKE SENSE: The returned intentResult is always void/an empty object. therefore its state never changes, either before or after the intent listener returns
```

Comment:
The issue is that you are using await. The point is that the response is a promise that then resolves. When you use await it waits for the promise to resolve so you are not seeing the state change and you are not seeing the promise. You are also not actually checking that it is a promise as await will pass through a synchronous response.

instead of this:
```javascript
let intentResult = await control.getIntentResult(intentResolution);`
```

you should be doing something like this:
```javascript
let intentResultPromise = control.getIntentResult(intentResolution);

//validate intentResultPromise is a promise (then-able)
if (typeof intentResultPromise?.then === 'function') {
    // probably a promise
} else {
    // definitely not a promise
}
`
Then await it for the result afterwards. Customize timeout based on timeout used in the app. Test is duplicated with longer timeouts to detect implementations that timeout too early.